src/urllib3/http2.py

```diff
@@ -97,11 +97,15 @@ def putrequest(
                 )
             )
 
-    def putheader(self, header: str, *values: str) -> None:
+    def putheader(self, header: str, *values: str) -> None:  # type: ignore[override]
         for value in values:
-            self._h2_headers.append(
-                (header.encode("utf-8").lower(), value.encode("utf-8"))
-            )
+            if isinstance(header, str):
```

Comment:
I'm surprising this isn't giving a type error, if `header` or `value` aren't strings then `header_bytes` and `value_bytes` aren't defined? Should we revert the change adding the `isinstance` checks?

Reply:
Remove is instance check and add type override